TWC_Backend/TWC_Backend/Controllers/ProjectController.cs

```diff
@@ -0,0 +1,138 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using TWC_DatabaseLayer.DTOs;
+using TWC_Services.Mapper;
+using TWC_DatabaseLayer.Models;
+using TWC_Services.DBService.Interfaces;
+using TWC_DatabaseLayer;
+using Microsoft.AspNetCore.Components.Forms;
+
+namespace TWC_Backend.Controllers
+{
+    [ApiController]
+    [Route("[controller]")]
+    public class ProjectController : Controller
+    {
+        private IDBProjectService _dbProjectService;
+        private IDBUserService _dbUserService;
+        private IMapper<Project, ProjectCreationDTO> _projectMapper;
+        private IDBTagService _dbTagService;
+
+        public ProjectController(
+            IDBProjectService dbProjectService,
+            IDBUserService dbUserService,
+            IMapper<Project, ProjectCreationDTO> projectMapper,
+            IDBTagService dBTagService
+            )
+        {
+            _dbProjectService = dbProjectService;
+            _projectMapper = projectMapper;
+            _dbUserService = dbUserService;
+            _dbTagService = dBTagService;
+        }
+
+        [HttpGet]
+        [Route("GetAllProjects")]
+        public async Task<ActionResult<List<Project>>> GetAllProjectsAsync()
+        {
+            try
+            {
+                return Ok(await _dbProjectService.GetAllProjectsAsync());
+            }
+            catch (Exception ex)
+            {
+                return BadRequest(ex.Message);
+            }
+        }
+
+        [HttpPost]
+        [Route("AddProject")]
+        public async Task<ActionResult<Project>> AddProject(ProjectCreationDTO project)
+        {
+
+            Project newProject = await _dbProjectService.CreateProjectAsync(project);
+
+            if (newProject == null)
+            {
+                return NotFound("Project or projects fields not found");
+            }
+
+            return Ok(newProject);
+        }
+
+        [HttpPut]
+        [Route("EditProject")]
+        public async Task<ActionResult<Project>> EditProject(ProjectEditDTO editProject)
+        {
+
+            Project edited = await _dbProjectService.EditProjectAsync(editProject);
+
+            if(edited == null) 
+            {
+                return NotFound("Project not found");
+            }
+
+            return Ok(await _dbProjectService.GetAllProjectsAsync());
```

Comment:
Isn't there a need for a try catch? Just if there is an error, the program will not crash?